File: frontend/src/services/requirementStudio.ts
import api from './api'

export interface StageStatus {
  stage_id: string
  stage_name: string
  status: string
  locked: boolean
  completed: boolean
  progress_percent: number
}

export interface RequirementStudioStatus {
  project_id: string
  stages: StageStatus[]
  current_stage_id: string | null
  overall_progress: number
}

export interface ArtifactSummary {
  artifact_id: string
  file_name: string
  file_type: string
  stage_id: string | null
  created_at: string | null
  content_preview?: string
  file_path?: string
}

export interface BaselinePayload {
  name: string
  description?: string
  stage_ids?: string[]
}

export interface StaleAnalysisResult {
  stale_artifacts: Array<{
    artifact_id: string
    artifact_name: string
    stage_id: string
    last_updated: string
    stale_reason: string
  }>
  summary: string
}

export interface ChangeRequestPayload {
  title: string
  description: string
  affected_artifacts: string[]
  target_stage_id: string
}

export async function fetchStudioStatus(projectId: string): Promise<RequirementStudioStatus> {
  const res = await api.get<RequirementStudioStatus>(`/v1/requirement-studio/${projectId}/status`)
  return res.data
}

export async function executeStage(projectId: string, stageId: string): Promise<void> {
  await api.post(`/v1/requirement-studio/${projectId}/stage/${stageId}/execute`)
}

export async function fetchArtifacts(projectId: string): Promise<ArtifactSummary[]> {
  const res = await api.get<ArtifactSummary[]>(`/v1/requirement-studio/${projectId}/artifacts`)
  return res.data
}

export async function createBaseline(projectId: string, payload: BaselinePayload): Promise<void> {
  await api.post(`/v1/requirement-studio/${projectId}/governance/baseline`, payload)
}

export async function fetchStaleAnalysis(projectId: string): Promise<StaleAnalysisResult> {
  const res = await api.get<StaleAnalysisResult>(
    `/v1/requirement-studio/${projectId}/governance/stale-analysis`,
  )
  return res.data
}

export async function createChangeRequest(
  projectId: string,
  payload: ChangeRequestPayload,
): Promise<void> {
  await api.post(`/v1/requirement-studio/${projectId}/governance/change-request`, payload)
}

File: backend/app/schemas/requirement_studio.py
"""Requirement Studio Pydantic schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class StageStatusItem(BaseModel):
    """Status item for a single stage in the requirement studio."""

    stage_id: str = Field(description="业务阶段标识")
    stage_name: str = Field(description="阶段名称")
    status: str = Field(description="UI 状态: locked | not_started | in_progress | review_pending | passed")
    progress: int = Field(description="进度百分比 0-100")
    can_enter: bool = Field(description="是否可进入该阶段")


class RequirementStudioStatusResponse(BaseModel):
    """Response for GET /{project_id}/status."""

    project_id: str
    current_stage: str
    stages: list[StageStatusItem]


class StageTaskItem(BaseModel):
    """Task-like item derived from a StageSkillBinding."""

    task_id: str
    task_name: str
    task_type: str
    status: str
    skill_id: str
    output_artifact: str | None = None


class StageTasksResponse(BaseModel):
    """Response for GET /{project_id}/stage/{stage_id}/tasks."""

    stage_id: str
    tasks: list[StageTaskItem]


class StageExecuteRequest(BaseModel):
    """Request for POST /{project_id}/stage/{stage_id}/execute."""

    skill_id: str | None = Field(default=None, max_length=36)
    context: dict[str, Any] | None = Field(default=None)
    reference_materials: list[str] | None = Field(default=None)


class StageExecuteResponse(BaseModel):
    """Response for stage execution trigger."""

    execution_id: str
    status: str


class StageReviewRequest(BaseModel):
    """Request for POST /{project_id}/stage/{stage_id}/review."""

    comments: list[str] | None = Field(default=None)
    suggestions: list[str] | None = Field(default=None)
    action: str = Field(..., description="pass | regenerate")


class StageReviewResponse(BaseModel):
    """Response for stage review submission."""

    stage_id: str
    status: str
    next_stage_id: str | None = None


class ArtifactFileItem(BaseModel):
    """Artifact file summary within a stage group."""

    artifact_id: str
    file_name: str
    version: str
    status: str


class StageArtifactGroup(BaseModel):
    """Artifacts grouped by stage."""

    stage_id: str
    stage_name: str
    files: list[ArtifactFileItem]


class ArtifactsResponse(BaseModel):
    """Response for GET /{project_id}/artifacts."""

    artifacts: list[StageArtifactGroup]


class ArtifactVersionItem(BaseModel):
    """Artifact version snapshot."""

    version: str
    created_at: str


class ArtifactContentResponse(BaseModel):
    """Response for GET /{project_id}/artifacts/{artifact_id}."""

    content: str
    versions: list[ArtifactVersionItem]


class ArtifactEditRequest(BaseModel):
    """Request for POST /{project_id}/artifacts/{artifact_id}/edit."""

    content: str
    version: str | None = Field(default=None)


class ArtifactEditResponse(BaseModel):
    """Response for artifact edit."""

    artifact_id: str
    version: str
    has_conflict: bool


class BaselineRequest(BaseModel):
    """Request for POST /{project_id}/governance/baseline."""

    artifact_ids: list[str]
    description: str | None = Field(default=None)


class BaselineResponse(BaseModel):
    """Response for baseline creation."""

    baseline_id: str
    version: str
    created_at: str


class StaleImpactItem(BaseModel):
    """Impact chain item for a stale artifact."""

    type: str
    target: str
    suggestion: str


class StaleArtifactItem(BaseModel):
    """Stale artifact detail."""

    artifact_id: str
    artifact_name: str
    version: str
    impact: list[StaleImpactItem]


class StaleAnalysisResponse(BaseModel):
    """Response for GET /{project_id}/governance/stale-analysis."""

    stale_artifacts: list[StaleArtifactItem]


class ChangeRequestRequest(BaseModel):
    """Request for POST /{project_id}/governance/change-request."""

    target_artifact_id: str
    change_type: str
    reason: str


class ChangeRequestResponse(BaseModel):
    """Response for change request creation."""

    change_request_id: str
    status: str
